go/private/rules/transition.bzl

```diff
@@ -290,25 +324,49 @@ def _go_reset_target_impl(ctx):
     )
     return providers
 
-go_reset_target = rule(
-    implementation = _go_reset_target_impl,
+go_exec_reset_target = rule(
+    implementation = _go_exec_reset_target_impl,
     attrs = {
         "dep": attr.label(
             mandatory = True,
-            cfg = go_reset_transition,
+            cfg = go_exec_reset_transition,
         ),
-        "_whitelist_function_transition": attr.label(
-            default = "@bazel_tools//tools/whitelists/function_transition_whitelist",
+        "_allowlist_function_transition": attr.label(
+            default = "@bazel_tools//tools/allowlists/function_transition_allowlist",
+        ),
+    },
+    doc = """Forwards providers from a target and applies go_exec_reset_transition.
+
+go_exec_reset_target depends on a single target, built using go_exec_reset_transition.
+It forwards Go providers and DefaultInfo.
+
+This is used to work around a problem with building tools: tools should be
+built with 'cfg = "exec"' so they work on the execution platform, but we also
+need to apply go_exec_reset_transition, so for example, a tool isn't built as a
+shared library with race instrumentation. This acts as an intermediately rule
+so we can apply both transitions.
+""",
+)
+
+go_non_go_exec_reset_target = rule(
+    implementation = _go_exec_reset_target_impl,
+    attrs = {
+        "dep": attr.label(
+            mandatory = True,
+            cfg = go_non_go_exec_reset_tool_transition,
+        ),
+        "_allowlist_function_transition": attr.label(
+            default = "@bazel_tools//tools/allowlists/function_transition_allowlist",
         ),
     },
-    doc = """Forwards providers from a target and applies go_reset_transition.
+    doc = """Forwards providers from a target and applies go_exec_reset_transition.
 
-go_reset_target depends on a single target, built using go_reset_transition.
+go_exec_reset_target depends on a single target, built using go_exec_reset_transition.
```

Comment:
This doc seems wrong

Reply:
I took the chance to improve the docs, but am not sure I understood this particular point.